python/lsst/pipe/tasks/makeCoaddTempExp.py

```diff
@@ -703,12 +749,63 @@ def prepareCalibratedExposures(self, calExpList, backgroundList=None, skyCorrLis
             Sequence of backgrounds to be added back in if bgSubtracted=False
         skyCorrList : `list` of `lsst.afw.math.backgroundList`
             Sequence of background corrections to be subtracted if doApplySkyCorr=True
+        externalSkyWcsCatalog : `lsst.afw.table.ExposureCatalog`
+            Exposure catalog with external skyWcs to be applied
+            if config.doApplyExternalSkyWcs=True.
+        externalPhotoCalibCatalog : `lsst.afw.table.ExposureCatalog`
+            Exposure catalog with external photoCalib to be applied
+            if config.doApplyExternalPhotoCalib=True.
         """
         backgroundList = len(calExpList)*[None] if backgroundList is None else backgroundList
         skyCorrList = len(calExpList)*[None] if skyCorrList is None else skyCorrList
+
+        if externalSkyWcsCatalog is not None:
+            externalSkyWcsDetectors = list(externalSkyWcsCatalog['detector_id'])
+        if externalPhotoCalibCatalog is not None:
+            externalPhotoCalibDetectors = list(externalPhotoCalibCatalog['detector_id'])
+        includeCalibVar = self.config.includeCalibVar
+
         for calexp, background, skyCorr in zip(calExpList, backgroundList, skyCorrList):
             mi = calexp.maskedImage
             if not self.config.bgSubtracted:
                 mi += background.getImage()
+
+            if externalSkyWcsCatalog is not None or externalPhotoCalibCatalog is not None:
+                detectorId = calexp.getInfo().getDetector().getId()
+
+            # Find and apply external photoCalib
+            if externalPhotoCalibCatalog is not None:
```

Comment:
Could you break out the external calibrations stuff into some short methods? e.g. `_getExternalPhotoCalib()` that returns the correct thing? I think that would tidy this up a bunch.

Reply:
Is that necessary? I don't see the need to create a separate function unless there is reason to believe that it will be used somewhere else, which doesn't seem likely in this case.

Reply:
I'm not going to split this into separate mini-functions, but the routine has been made simpler by the use of `catalog.find(detectorId)` with the new format.